addons/l10n_in_edi_ewaybill/models/account_edi_format.py

```diff
@@ -0,0 +1,601 @@
+                    odoobot = self.env.ref("base.partner_root")
+                    invoices.message_post(author_id=odoobot.id, body=_(
+                        "Somehow this E-waybill is generate in government portal before." \
+                        "<br/>Normally, this should not happen too often" \
+                        "<br/>Just verify value by enter details into government website " \
+                        "<a href='https://ewaybillgst.gov.in/Others/EBPrintnew.aspx'>here<a>."
+                    ))
+            if "no-credit" in error_codes:
+                res[invoices] = {
+                    "success": False,
+                    "error": self._l10n_in_edi_get_iap_buy_credits_message(invoices.company_id),
+                    "blocking_level": "error",
+                }
+            elif error:
+                error_message = "<br/>".join(["[%s] %s" % (e.get("code"), html_escape(e.get("message"))) for e in error])
+                blocking_level = "error"
+                if "404" in error_codes or "waiting" in error_codes:
+                    blocking_level = "warning"
+                res[invoices] = {
+                    "success": False,
+                    "error": error_message,
+                    "blocking_level": blocking_level,
+                }
+        if not response.get("error"):
+            json_dump = json.dumps(response.get("data"))
+            json_name = "%s_irn_ewaybill.json" % (invoices.name.replace("/", "_"))
+            attachment = self.env["ir.attachment"].create({
+                "name": json_name,
+                "raw": json_dump.encode(),
+                "res_model": "account.move",
+                "res_id": invoices.id,
+                "mimetype": "application/json",
+            })
+            inv_res = {"success": True, "attachment": attachment}
+            res[invoices] = inv_res
+        return res
+
+    def _l10n_in_edi_irn_ewaybill_generate_json(self, invoice):
+        json_payload = {
+            "Irn": invoice._get_l10n_in_edi_response_json().get("Irn"),
+            "Distance": invoice.l10n_in_distance,
+        }
+        if invoice.l10n_in_mode == "0":
+            json_payload.update({
+                "TransId": invoice.l10n_in_transporter_id.vat,
+                "TransName": invoice.l10n_in_transporter_id.name,
+            })
+        elif invoice.l10n_in_mode == "1":
+            json_payload.update({
+                "TransMode": invoice.l10n_in_mode,
+                "VehNo": invoice.l10n_in_vehicle_no,
+                "VehType": invoice.l10n_in_vehicle_type,
+            })
+        elif invoice.l10n_in_mode in ("2", "3", "4"):
+            doc_date = invoice.l10n_in_transportation_doc_date
+            json_payload.update({
+                "TransMode": invoice.l10n_in_mode,
+                "TransDocDt": doc_date and doc_date.strftime("%d/%m/%Y") or False,
+                "TransDocNo": invoice.l10n_in_transportation_doc_no,
```

Comment:
```suggestion
                "TransDocNo": invoice.l10n_in_transportation_doc_no[-15:],
```

Reply:
we don't trim it because the user doesn't know that we trip it so when the user is entering the details then he trimmed it by himself.

Reply:
still, do you think we trim it?

Reply:
yes, but we send what is in odoo so not trimmed